### ISSUE TITLE
Is anybody using the re-designed router?

### ISSUE THREAD

**Issue by romansky** [japgolly/scalajs-react#239]
\<rant\>I've been using the old router in a side project I was working on and now came back to, the new API's and specifically the router API's are IMHO in very bad taste, lots of "sugar" Scala sprinkled on top, renders IDE useless, strange / bad naming conventions, lacking coherent documentation..... ..
Unfortunately I feel very driven to not use this project any more and may have to "roll my own" soon..
sigh... \</rant\>

Bottom line: was looking around how to know when a route changed, an hour later-still looking..!

**Comment by japgolly** [comment]
Lacking coherent documentation? What are you talking about? There's a
whole, big page dedicated to nothing but the router! Did you miss that?

I can't find any value in your other criticisms either.

> bad taste
Too vague.

> Scala sugar.
It's a Scala project.

> IDE useless.
Complain to IDE makes, not me. It's valid Scala.

> Strange names
Subjective and not very important anyway. If you think the names you come
up with are beautiful and clear you'll like find users elsewhere that will
find them ambiguous and strange too. Being unambiguous, clear, deceptive
and concise is very, very hard.

Rolling your own and duplicating all this logic cos you don't like the
*aesthetics* of the current router would be silly and immature. Would it
not be a better idea to offer up your opinion on how you think the
aesthetics should be, and improve what we already have?

**Comment by japgolly** [comment]
@romansky I see you've updated your comment saying that what you actually want to know is how to detect when a route changes? If you provide some more details about why/when you want to know and what you want to do with the info, I should be able to provide you an answer.
